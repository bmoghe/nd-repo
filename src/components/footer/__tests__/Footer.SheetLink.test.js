import React from 'react';
import { shallow } from "enzyme";
import { Sheet } from "../Footer.SheetLink";
import toJson from 'enzyme-to-json';

describe('Footer SheetLink Component', () => {
  it('should not crash', () => {
    const props = {
      sheet: {
        name: 'test'
      }
    };
    const wrapper = shallow(<Sheet {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should display correct number of sheets', () => {
    const props = {
      sheet: {
        name: 'test'
      }
    };
    const wrapper = shallow(<Sheet {...props} />);
    expect(wrapper).not.doBe(undefined);
  });

  it('should be selected default sheet', () => {
    const props = {
      sheet: {
        name: 'test'
      }
    };
    const wrapper = shallow(<Sheet {...props} />);
  });

  it('should open action popup to make duplicate/delete/set default', () => {
    
  });

  it('should navigate to "sheet/:id" upon click, validate the url', () => {
    
  });
})
