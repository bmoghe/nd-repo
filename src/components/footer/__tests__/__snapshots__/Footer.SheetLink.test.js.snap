// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer SheetLink Component should not crash 1`] = `
<div
  className="nav-item "
  id="menuitemPopoverundefined"
  onClick={[Function]}
  title="test"
>
  <span
    className="footer-icon sheet-icon"
    id="menuitemIcon"
  />
  test
  <Popover
    content={
      <div
        id="SHEET_MENU"
      >
        <button
          className="btn btn-link cursor-pointer item-disable"
          onClick={[Function]}
          type="button"
        >
           Delete 
        </button>
        <br />
        <button
          className="btn btn-link cursor-pointer"
          onClick={[Function]}
          type="button"
        >
           Duplicate
        </button>
        <br />
        <button
          className="btn btn-link cursor-pointer"
          onClick={[Function]}
          type="button"
        >
           Mark as default
        </button>
        <br />
      </div>
    }
    mouseEnterDelay={0.1}
    mouseLeaveDelay={0.1}
    onVisibleChange={[Function]}
    overlayClassName="menu-popover"
    overlayStyle={Object {}}
    placement="topLeft"
    transitionName="zoom-big"
    trigger="click"
    visible={false}
  />
  <span
    className="solid-budicon-chevron-bottom pop-btn"
    onClick={[Function]}
  />
  <Modal
    closable={false}
    destroyOnClose={true}
    disableBackdropClick={false}
    footer={null}
    maskStyle={
      Object {
        "backgroundColor": "rgba(92, 100, 118, 0.5)",
      }
    }
    open={false}
    width={600}
    wrapClassName="data-src-confirm-modal"
  >
    <div>
      <div
        className="ae-modal-header"
      >
        <div
          className="ae-header-title"
        >
          Delete data Sheet
        </div>
        <button
          autoFocus={true}
          className="ae-header-close"
          onClick={[Function]}
          type="button"
        />
      </div>
      <div
        className="msg"
      >
        <span>
          Are you sure to delete
           
          <b>
            "
            test
            "
          </b>
           ?
        </span>
      </div>
      <div
        className="confirmation"
      >
        <Button
          onClick={[Function]}
          title="No"
        />
        <Button
          onClick={[Function]}
          title="Yes"
          type="primary"
        />
      </div>
    </div>
  </Modal>
</div>
`;
